doc/gui/examples/controls/chat-streaming.py

```diff
@@ -0,0 +1,192 @@
+            append_to_message(state, f"\n* {event.get('year', '')}: {event.get('text', '')}", idx)
+            # invoke update_message_with_image in a thread
+            invoke_long_callback(
+                state=state,
+                user_function=update_message_with_image,
+                user_function_args=[
+                    state.get_gui(),
+                    get_state_id(state),
+                    idx,
+                    event.get("text", ""),
+                    pages[0].get("thumbnail", {}) if (pages := event.get("pages", [])) and len(pages) else {},
+                ],
+            )
+
+
+# set or append a message returning its index in the list
+def set_message(state: State, message: str, idx: t.Optional[int] = None):
+    if idx is not None and idx < len(state.messages):
+        msg = state.messages[idx]
+        state.messages[idx] = (msg[0], message, msg[2])
+    else:
+        idx = len(state.messages)
+        state.messages.append((f"{len(state.messages)}", message, users[0][0]))
+    state.refresh("messages")
+    return idx
+
+
+# append to an existing message
+def append_to_message(state: State, message: str, idx: int):
+    if idx < len(state.messages):
+        msg = state.messages[idx]
+        state.messages[idx] = (msg[0], f"{msg[1]}{message}", msg[2])
+        state.refresh("messages")
+    return idx
+
+
+# request the wikipedia API, invoked by send_message
+def request_wikipedia(gui: Gui, state_id: str, event_type: str, month: str, day: str):
+    # inform the user a request is posted
+    idx = invoke_callback(
+        gui,
+        state_id,
+        set_message,
+        ["requesting Wikipedia ..."],
+    )
+    request = wiki_url.format(type=event_type, month=month, day=day)
+    req = requests.get(request, headers={"accept": "application/json; charset=utf-8;", "User-Agent": user_agent})
+    # handle the response
+    if req.status_code == 200:
+        # display response
+        invoke_callback(
+            gui,
+            state_id,
+            update_message,
+            [req.json(), event_type, f"{day}/{month}", idx],
+        )
+    else:
+        # display error
+        invoke_callback(
+            gui,
+            state_id,
+            set_message,
+            [f"requesting Wikipedia failed: {req.status_code}", idx],
+        )
+
+
+# invoked by the on_action on the chat control when the user press the send button
+def send_message(state: State, id: str, payload: dict):
+    args = payload.get("args", [])
+
+    # display request
+    state.messages.append((f"{len(state.messages)}", args[2], args[3]))
+    state.refresh("messages")
+
+    # analyse request
+    request = args[2].lower()
+    type_event = None
+    for word in event_types:
+        if word in request:
+            type_event = event_types[word]
+            break
+    type_event = type_event if type_event else "events"
+
+    month = None
+    day = None
+    for m in re.finditer(r"(\d\d?)", request):
```

Comment:
```suggestion
    for m in re.finditer(r"(\d+)", request):
```
Shorted and less constrained... may be not what you really want.

Reply:
1 or 2 digits